x-pack/test/api_integration/deployment_agnostic/apis/observability/dataset_quality/utils/dataset_quality_user.ts

```diff
@@ -0,0 +1,182 @@
+ * @param fallbackRole
+ */
+export async function getDatasetQualityNoAccessSuperTestUser({
+  getService,
+  requestHeaderOptions = {
+    useCookieHeader: true,
+    withInternalHeaders: true,
+  },
+  fallbackRole,
+}: DatasetQualitySuperUserParams): Promise<DatasetQualitySupertestUser> {
+  const noAccessUserRoleDescriptors: KibanaRoleDescriptors = {
+    elasticsearch: {
+      indices: [],
+    },
+    kibana: [],
+  };
+
+  return getDatasetQualitySuperTestUserForRoleDescriptors(
+    { getService, requestHeaderOptions },
+    noAccessUserRoleDescriptors,
+    fallbackRole
+  );
+}
+
+/**
+ * Returns a Role Scoped Supertest User corresponding to the provided role descriptors.
+ *
+ * @Note: Falls back to Admin Supertest User if custom role is not available.
+ *
+ * @param getService { DeploymentAgnosticFtrProviderContext['getService'] } The getService from the FTR Provider Context
+ * @param requestHeaderOptions { RequestHeadersOptions } The request header options
+ * @param descriptors { KibanaRoleDescriptors } The role descriptors for the custom role
+ * @param fallbackRole { 'admin' | 'editor' | 'viewer' } The fallback role if custom role is not available
+ * @returns { user: SupertestWithRoleScopeType, isCustomRoleEnabled: boolean, clean: () => Promise<void> } The user,
+ * a flag indicating if the custom role is enabled, and a clean function to clean up the user after the test
+ */
+async function getDatasetQualitySuperTestUserForRoleDescriptors(
+  {
+    getService,
+    requestHeaderOptions = {
+      useCookieHeader: true,
+      withInternalHeaders: true,
+    },
+  }: Pick<DatasetQualitySuperUserParams, 'getService' | 'requestHeaderOptions'>,
+  descriptors: KibanaRoleDescriptors,
+  fallbackRole: DatasetQualitySuperUserParams['fallbackRole'] = 'admin'
+): Promise<DatasetQualitySupertestUser> {
+  const roleScopedSupertest = getService('roleScopedSupertest');
+  const samlAuth = getService('samlAuth');
+  const config = getService('config');
+  const authRoleProvider = getAuthProvider({ config });
+  const isCustomRoleEnabled = authRoleProvider.isCustomRoleEnabled();
+
+  // Fallback to Admin Supertest User if custom role is not available
+  // TODO: Remove this fallback once custom role is available in all environments
+  if (!isCustomRoleEnabled) {
+    const user = await roleScopedSupertest.getSupertestWithRoleScope(
+      fallbackRole,
+      requestHeaderOptions
+    );
+    const clean = async () => {
+      await user.destroy();
+    };
+
+    return { user, isCustomRoleEnabled: false, clean };
+  }
+
+  // Define the custom role for the Dataset Quality Monitor User
+  await samlAuth.setCustomRole(descriptors);
+  const user = await roleScopedSupertest.getSupertestWithRoleScope(
+    samlAuth.getCustomRole(),
+    requestHeaderOptions
+  );
+  const clean = async () => {
+    await user.destroy();
+    await samlAuth.deleteCustomRole();
+  };
+
+  return { user, isCustomRoleEnabled: true, clean };
+}
```

Comment:
I think we already had a chat about it, but I will repeat myself: 

Deployment-agnostic tests target cloud-first testing and currently custom roles for Oblt project are disabled.
It means that any Oblt serverless feature requiring custom roles doesn't meet requirements for "Deployment-Agnostic tests", because in ESS deployment we can create custom roles.

Running the same test with different roles in serverless and stateful as part of DO simply _breaks the concept_ and will complicate investigating the failures. These tests are part of Release quality gates and its logic must be "clean"(no logic split in terms of authentication as it might lead to missing privileges bugs in Production). 

https://github.com/elastic/kibana/tree/main/x-pack/test/api_integration/deployment_agnostic#when-to-create-separate-tests

`Role-Specific Logic: If API access or logic depends on roles that differ across deployments.`

All the custom roles related testing for Oblt serverless project should be loaded in https://github.com/elastic/kibana/blob/main/x-pack/test_serverless/api_integration/test_suites/observability/config.feature_flags.ts where we enabled custom roles.